src/test/java/uk/gov/hmcts/darts/arm/rpo/impl/ArmRpoApiSaveBackgroundSearchTest.java

```diff
@@ -1,22 +1,149 @@
 import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.hmcts.darts.arm.client.ArmRpoClient;
+import uk.gov.hmcts.darts.arm.client.model.rpo.SaveBackgroundSearchRequest;
+import uk.gov.hmcts.darts.arm.client.model.rpo.SaveBackgroundSearchResponse;
+import uk.gov.hmcts.darts.arm.config.ArmApiConfigurationProperties;
+import uk.gov.hmcts.darts.arm.exception.ArmRpoException;
+import uk.gov.hmcts.darts.arm.helper.ArmRpoHelper;
+import uk.gov.hmcts.darts.arm.helper.ArmRpoHelperMocks;
+import uk.gov.hmcts.darts.arm.service.ArmRpoService;
+import uk.gov.hmcts.darts.common.entity.ArmRpoExecutionDetailEntity;
+import uk.gov.hmcts.darts.common.entity.UserAccountEntity;
 
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
 import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
 
 @ExtendWith(MockitoExtension.class)
 class ArmRpoApiSaveBackgroundSearchTest {
 
+    @Mock
+    private ArmRpoClient armRpoClient;
+
+    @Mock
+    private ArmRpoService armRpoService;
+
+    @Mock
+    private ArmRpoHelper armRpoHelper;
+
+    @Mock
+    private ArmApiConfigurationProperties armApiConfigurationProperties;
+
     @InjectMocks
     private ArmRpoApiImpl armRpoApi;
 
+    private UserAccountEntity userAccount;
+    private static final Integer EXECUTION_ID = 1;
+    private static final ArmRpoHelperMocks ARM_RPO_HELPER_MOCKS = new ArmRpoHelperMocks();
+
+
+    @BeforeEach
+    void setUp() {
+        userAccount = new UserAccountEntity();
+
+        ArmRpoExecutionDetailEntity armRpoExecutionDetailEntity = new ArmRpoExecutionDetailEntity();
+        armRpoExecutionDetailEntity.setId(EXECUTION_ID);
+        when(armRpoService.getArmRpoExecutionDetailEntity(EXECUTION_ID)).thenReturn(armRpoExecutionDetailEntity);
+    }
+
+    @Test
+    void saveBackgroundSearchReturnsSuccess() {
+        // given
+        SaveBackgroundSearchResponse saveBackgroundSearchResponse = new SaveBackgroundSearchResponse();
+        saveBackgroundSearchResponse.setStatus(200);
+        when(armRpoClient.saveBackgroundSearch(anyString(), any())).thenReturn(saveBackgroundSearchResponse);
+
+        // when
+        armRpoApi.saveBackgroundSearch("token", 1, "searchName", userAccount);
+
+        // then
+        verify(armRpoService).updateArmRpoStateAndStatus(any(),
+                                                         eq(ARM_RPO_HELPER_MOCKS.getSaveBackgroundSearchRpoState()),
+                                                         eq(ARM_RPO_HELPER_MOCKS.getInProgressRpoStatus()),
+                                                         any());
+        verify(armRpoService).updateArmRpoStatus(any(), eq(ARM_RPO_HELPER_MOCKS.getCompletedRpoStatus()), any());
+    }
```

Comment:
Consider adding a `verifyNoMoreInteractions(armRpoService);` at the end to verify there is no unexpected 3rd interaction after moving the status to complete (same comment applies to all tests here).